src/commands/apex/get/test.ts

```diff
@@ -63,25 +63,31 @@ export default class Test extends SfCommand<RunResult> {
   };
 
   public async run(): Promise<RunResult> {
-    const { flags } = await this.parse(Test);
+    try {
+      const { flags } = await this.parse(Test);
 
-    // add listener for errors
-    process.on('uncaughtException', (err) => {
-      throw messages.createError('apexLibErr', [err.message]);
-    });
+      const conn = flags['target-org'].getConnection(flags['api-version']);
 
-    const conn = flags['target-org'].getConnection(flags['api-version']);
+      const testService = new TestService(conn);
+      const result = await testService.reportAsyncResults(flags['test-run-id'], flags['code-coverage']);
 
-    const testService = new TestService(conn);
-    const result = await testService.reportAsyncResults(flags['test-run-id'], flags['code-coverage']);
+      const testReporter = new TestReporter(new Ux({ jsonEnabled: this.jsonEnabled() }), conn, this.config.bin);
 
-    const testReporter = new TestReporter(new Ux({ jsonEnabled: this.jsonEnabled() }), conn, this.config.bin);
-
-    return testReporter.report(result, {
-      'output-dir': flags['output-dir'],
-      'result-format': flags['result-format'],
-      json: flags.json,
-      'code-coverage': flags['code-coverage'],
-    });
+      return await testReporter.report(result, {
+        'output-dir': flags['output-dir'],
+        'result-format': flags['result-format'],
+        json: flags.json,
+        'code-coverage': flags['code-coverage'],
+      });
+    } catch (e) {
+      if (e instanceof SfError) {
+        // something we threw
+        throw e;
+      } else {
+        // what used to be caught by the
+        // process.on('uncaughtException', err => {
+        throw SfError.wrap(messages.createError('apexLibErr', [(e as Error).message]));
+      }
+    }
```

Comment:
the try/catch wrapper can be removed, `sfCommand.catch` handles all errors here:
https://github.com/salesforcecli/sf-plugins-core/blob/main/src/sfCommand.ts#L402C1-L408C7

what if we ask apex-node to use `SfError` instead of catching everything here?

https://github.com/search?q=repo%3Aforcedotcom%2Fsalesforcedx-apex+%2Fthrow+%28new+Error%7CError%29%5C%28%2F&type=code

that way the CLI get better errors too (IDEx team can add `error.actions` if applicable, we preserve their original `error.name`,etc).